### ISSUE TITLE
Fails to strip comments on angular.js ("*/*" inside comment)

### ISSUE THREAD

**Issue by atomantic** [RnbWd/gulp-strip-comments#3]
Angular.js has an section containing:

```
common: {
        'Accept': 'application/json, text/plain, */*'
      },
```

This causes a strip issue. If I change angular.js to contain:

```
 'Accept': 'application/json, text/plain, *' + '/' + '*'
```

it works :)

**Comment by RnbWd** [comment]
Sorry for the late response. I typically advocate posting issues like this in the [strip-comments](https://github.com/jonschlinkert/strip-comments) repo (which this library is a wrapper) ... but it doesn't look like that library is being actively maintained, so I've considered forking it or using an alternative method for the last few months.

1) On the plus side - `strip-comments` is very simple and most of the logic is contained in these regexes:

```js
var reBlock = /\/\*(?!\/)(.|[\r\n]|\n)+?\*\/\n?\n?/gm
var reBlockIgnore = /\/\*(?!(\*?\/|\*?\!))(.|[\r\n]|\n)+?\*\/\n?\n?/gm
var reLine = /(^|[^\S\n])(?:\/\/)([\s\S]+?)$/gm;
var reLineIgnore = /(^|[^\S\n])(?:\/\/[^!])([\s\S]+?)$/gm;
```

- So if there's an easy fix, I'll will fork and maintain that file here. Just let me know if you can figure out the issue [regex101](https://regex101.com)

2) There's also [strip-json-comments](https://github.com/sindresorhus/strip-json-comments) (with a corresponding gulp plugin) maintained by @sindresorhus who is prolific (IMO) - I'm not sure if it works with all js code (I've never tried, it's made for JSON) - but it's a very similar implementation. 

3) Using AST parsers / transformers like `esprima`, and `acorn` and` babel` - which is a lot of overhead for something this simple and would make more sense as a [babel-plugin](https://github.com/babel-plugins).

Basically - this library contains no code for stripping comments - so there's nothing I can do without forking or reimplementing what already exists. So i'll keep this comment open in case someone smarter than myself (or better with regexp) knows the solution to this problem, and I will more closely at `strip-json-comments` which may contain the solution to this problem.

**Comment by atomantic** [comment]
thanks for the response. We found a work around with how we were building/bundling angular to resolve it. If I hit this issue again, I'll look to strip-comments for a fix :)

**Comment by vitaly-t** [comment]
Just for the record, [decomment](https://github.com/vitaly-t/decomment) doesn't get stumped by that example, it works correctly.
